pkg/provider/github/repository.go

```diff
@@ -0,0 +1,110 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-github/v47/github"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/v1alpha1"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/formatting"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params/clients"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/templates"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ConfigureRepository(ctx context.Context, run *params.Run, req *http.Request, payload string, logger *zap.SugaredLogger) (bool, bool, error) {
+	// gitea set x-github-event too, so skip it for the gitea driver
+	if h := req.Header.Get("X-Gitea-Event-Type"); h != "" {
+		return false, false, nil
+	}
+	event := req.Header.Get("X-Github-Event")
+	if event != "repository" {
+		return false, false, nil
+	}
+
+	eventInt, err := github.ParseWebHook(event, []byte(payload))
+	if err != nil {
+		return true, false, err
+	}
+	_ = json.Unmarshal([]byte(payload), &eventInt)
+	repoEvent, _ := eventInt.(*github.RepositoryEvent)
+
+	if repoEvent.GetAction() != "created" {
```

Comment:
Are you sure that you want filter only on created, isn't that what is issued too on issue comments or pullrequest create?

Reply:
we have already filtered out the event `eventInt.(*github.RepositoryEvent)`
we are filtering the action here